src/ext/php5/startup_logging.c

```diff
@@ -0,0 +1,435 @@
+#include "startup_logging.h"
+
+#include <SAPI.h>
+#include <Zend/zend_API.h>
+#include <curl/curl.h>
+#include <php.h>
+#include <stdbool.h>
+#include <time.h>
+
+#include <ext/standard/info.h>
+#include <ext/standard/php_smart_str.h>
+
+#include "coms.h"
+#include "configuration.h"
+#include "logging.h"
+#include "version.h"
+
+#define ISO_8601_LEN (20 + 1)  // +1 for terminating null-character
+
+// True global; only modify on MINIT/STARTUP
+static time_t startup_time = -1;
+
+static void _dd_get_startup_time(char *buf) {
+    time_t now = (startup_time == -1) ? time(NULL) : startup_time;
+    struct tm *tm = gmtime(&now);
+    if (tm) {
+        strftime(buf, ISO_8601_LEN, "%Y-%m-%dT%TZ", tm);
+    } else {
+        ddtrace_log_debug("Error getting startup time");
+    }
+}
+
+static void _dd_add_assoc_string(HashTable *ht, const char *name, size_t name_len, const char *str) {
+    zval *value;
+    MAKE_STD_ZVAL(value);
+    size_t str_len = str ? strlen(str) : 0;
+    if (str_len > 0) {
+        ZVAL_STRINGL(value, str, str_len, 1);
+    } else {
+        ZVAL_NULL(value);
+    }
+    zend_symtable_update(ht, name, (name_len + 1), (void **)&value, sizeof(zval *), NULL);
+}
+
+static void _dd_add_assoc_string_free(HashTable *ht, const char *name, size_t name_len, char *str) {
+    _dd_add_assoc_string(ht, name, name_len, (const char *)str);
+    free(str);
+}
+
+static void _dd_add_assoc_bool(HashTable *ht, const char *name, size_t name_len, bool v) {
+    zval *value;
+    MAKE_STD_ZVAL(value);
+    ZVAL_BOOL(value, v);
+    zend_symtable_update(ht, name, (name_len + 1), (void **)&value, sizeof(zval *), NULL);
+}
+
+static void _dd_add_assoc_double(HashTable *ht, const char *name, size_t name_len, double num) {
+    zval *value;
+    MAKE_STD_ZVAL(value);
+    ZVAL_DOUBLE(value, num);
+    zend_symtable_update(ht, name, (name_len + 1), (void **)&value, sizeof(zval *), NULL);
+}
+
+static char *_dd_get_ini(const char *name, size_t name_len) { return zend_ini_string((char *)name, (name_len + 1), 0); }
+
+static bool _dd_ini_is_set(const char *name, size_t name_len) {
+    const char *ini = _dd_get_ini(name, name_len);
+    return ini && (strcmp(ini, "") != 0);
+}
+
+// Modified version of zend_ini_parse_bool()
+// @see https://github.com/php/php-src/blob/28b4761/Zend/zend_ini.c#L493-L502
+static bool _dd_parse_bool(const char *name, size_t name_len) {
+    const char *ini = _dd_get_ini(name, name_len);
+    size_t ini_len = strlen(ini);
+    if ((ini_len == 4 && strcasecmp(ini, "true") == 0) || (ini_len == 3 && strcasecmp(ini, "yes") == 0) ||
+        (ini_len == 2 && strcasecmp(ini, "on") == 0)) {
+        return 1;
+    } else {
+        return atoi(ini) != 0;
+    }
+}
+
+static void _dd_get_agent_url(char *buf) {
+    char *host = get_dd_agent_host();
+    sprintf(buf, "https://%s:%d", host, (int)get_dd_trace_agent_port());
```

Comment:
Are you about http vs https here?

Reply:
Nice catch. This is actually supposed to be `http` and I updated the PR. Once #926 is merged, we could expose a public function to generate the URL for consistency.